### DISCUSSION TITLE
More options for Note Revisions handing

### DISCUSSION THREAD

**Discussion by Br0kenSilos** [TriliumNext/Notes#258] [Ideas]
I would love to see some additional options for handling note revisions.  For instance:

 - The ability to limit the number of note revisions to a specific (user selected) number and roll off the oldest revisions after that number is reached. So for instance, limit a particular branch, or note, or even the entire tree to have no more than say 3 revisions per note. When the time comes for a 4th revision, the oldest revision is then deleted in a FIFO order. The number of revisions should be flexible. So if perhaps I generally want the entire tree to have no more than 3 revisions per note, but I have one note (or branch) down in the tree somewhere that I would like to keep 10 revisions, it would be nice to be able override the 3 revisions set higher in the tree for that particular note/branch.
 
  - The option at the top of the tree (or perhaps under Options > Advanced Other (near the vacuum db option) ) to remove ALL current Revision Snapshots. This would be to simply shrink the DB and clear unused space when things are in a state that the user feels is safe to remove all snapshots.

**Comment by eliandoran** [comment]
@Br0kenSilos , those are really good ideas. I suggest you turn them into issues/feature requests.

For the second point, I was thinking maybe we need some kind of "usage analyzer". To show how much storage is occupied in total by the DB, by notes, by revisions, by attachments and by backups. And maybe offer some ways to reduce the consumption.

**Reply by Br0kenSilos** [reply]
I will put them in as feature requests. And I really like your idea for storage analytics. I know for me personally, I like to keep my DB as small as possible even though I have ample storage space available to that container on the that docker VM. The reasoning is that I dont want to have to monitor the partition on that VM constantly. I have several VM's and containers in multiple locations and trying to monitor all of it is a bit of a pita. But this is the very reason that I tend to not attach images to my notes unless they are just absolutely needed. Some storage analytics like you mentioned would be very helpful for me to quickly check how much storage my Trilium(Next) is taking up. Maybe even have an option to display the total storage currently used in the tool bar or something so its always available at a glance. Just tossing the idea out there.  :-)

**Reply by eliandoran** [reply]
@Br0kenSilos , on my side I love attaching things to my Trilium DB, it has recently grown to 700 MB. 🙈 

I'll try to think of a design for the storage analyzer.

**Reply by Br0kenSilos** [reply]
@eliandoran I hear ya. My DB is only 29MB at the moment because I refuse to add images unless I just _have_ to. It's got quite a bit in it but clearly its almost entirely text.  That being said, I would love to start adding more screenshots as that would make my notes more usable for sure. A picture is worth a thousand words!

However, I am so paranoid about losing my data (any data - but Especially my Trilium data)  that I run nightly backups of my DB and keep the last 64 backups on the drive itself. Then of course I have the entire VM (Which houses other containers as well), backed up to my NAS in a similar rotating fashion. Then the files are pulled from the NAS once a month or so and put on external drives that go in fire safe. On top of that, even though I usually just access my instance over the web UI (usually from multiple places at the same time) I also run the Desktop version on my desktop and laptop and usually open it once ever few days just to sync it. Its WAY overkill and its a "me" thing for sure.  Yes I love my Trilium that much!!

**Reply by eliandoran** [reply]
@Br0kenSilos , that's a really good redundancy. 💪🏻

**Comment by eliandoran** [comment]
Closing since we now have issue tracking them: #263, #262
